### PR TITLE
Fix for #270, and framework for testing cross-targeting scenarios

### PR DESCRIPTION
**Pull request by latkin**
ref #270 - "internal error: null: convTypeRefAux"

More issues with comma escaping, similar to #250.  Fix is pretty simple, just replace `,` with `\,` in a couple more places that need it. Also needed some small logic to make sure we don't *double*-escape when depickling names that are already escaped.

This is risky mostly in multi-targeting scenarios: 4.0 compiler escapes type names differently than 3.X compiler -- can each understand the output of the other?

Added a general-purpose mini test framework for validating such things, which cycles through various cross-version scenarios and makes sure everything plays nicely.

### CODE REVIEW

File: src/absil/ilreflect.fs [hunk 1]
```diff
@@ -324,25 +324,7 @@ type cenv =
 // [ns]            ,name -> ns+name
 // [ns;typeA;typeB],name -> ns+typeA+typeB+name
 let convTypeRefAux (cenv:cenv) (tref:ILTypeRef) = 
-
-    // If an inner nested type's name contains a space, the proper encoding is "\+" on both sides - otherwise,
-    // we use "+"
```
Comment (latkin) [inline src/absil/ilreflect.fs]: As far as I can tell from my testing, this comment is just *wrong* -- I can't find any scenarios where the + is escaped. Indeed, the incorrect escaping needed to be removed in order to get my tests passing. All current tests continue to work fine.

This is now simplified and exactly matches https://github.com/Microsoft/visualfsharp/blob/fsharp4/src/absil/il.fs#L682

**Comment by latkin** [comment]
For reference, some CLR code that pertains to parsing special chars in reflection-style type names (hooray open source :tada:)

https://github.com/dotnet/corefx/blob/dev/metadata/src/System.Reflection.Metadata/src/System/Reflection/Metadata/Decoding/TypeNameParsing/TypeNameParserOfT.cs

https://github.com/dotnet/coreclr/blob/cbf46fb0b6a0b209ed1caf4a680910b383e68cba/src/vm/typeparse.h#L28

Of the designated reserved characters, only `,` is allowed in F# backticked type names (and none are allowed in C#/VB type names)

**Comment by KevinRansom** [comment]
Looks good.  Nice work.

File: src/absil/ilreflect.fs (hunk 1 above)
Comment (dsyme) [inline src/absil/ilreflect.fs]: That code had always been very, very suspect to me.  Thanks for digging into this so thoroughly

**Pull request closed without merging**